apis/field_error.go

```diff
@@ -30,29 +30,28 @@ const CurrentField = ""
 
 // DiagnosticLevel is used to signal the severity of a particular diagnostic
 // in the form of a FieldError.
-type DiagnosticLevel string
+type DiagnosticLevel int
 
 const (
 	// ErrorLevel is used to signify fatal/blocking diagnostics, e.g. those
 	// that should block admission in a validating admission webhook.
-	ErrorLevel DiagnosticLevel = "Error"
+	ErrorLevel DiagnosticLevel = iota
 
 	// WarningLevel is used to signify information/non-blocking diagnostics,
 	// e.g. those that should be surfaced as warnings in a validating admission
 	// webhook.
-	WarningLevel DiagnosticLevel = "Warning"
+	WarningLevel
 )
 
-// Matches checks whether the provided diagnostic level matches this one,
-// including the special case where an empty DiagnosticLevel equals ErrorLevel.
-func (l DiagnosticLevel) Matches(r DiagnosticLevel) bool {
-	switch {
-	case l == ErrorLevel && r == "":
-		return true
-	case r == ErrorLevel && l == "":
-		return true
+func (dl DiagnosticLevel) String() string {
+	switch dl {
+	case ErrorLevel:
+		return "Error"
+	case WarningLevel:
+		return "Warning"
+
 	default:
-		return l == r
+		return fmt.Sprintf("<UNKNOWN: %d>", dl)
 	}
 }
```

Comment:
I think using `//go:generate stringer -type=DiagnosticLevel` will generate this code.

Reply:
I didn't do this because I wasn't sure it was worth adding `go generate` to the mix of things folks have to do 🤷 

I also figured it would do something like `"ErrorLevel"`, and `"WarningLevel"`, and I have a mild preference for this.

Neither is particular strong, but I also don't think it's saving us too much code either.